src/parser/jobs/brd/modules/ApexArrow.tsx

```diff
@@ -0,0 +1,108 @@
+import React from 'react'
+import {Trans, Plural, NumberFormat} from '@lingui/react'
+import {t} from '@lingui/macro'
+import {ActionLink} from 'components/ui/DbLink'
+import ACTIONS from 'data/ACTIONS'
+import {DamageEvent} from 'fflogs'
+import Module, {dependency} from 'parser/core/Module'
+import {NormalisedDamageEvent} from 'parser/core/modules/NormalisedEvents'
+import Suggestions, {TieredSuggestion, SEVERITY} from 'parser/core/modules/Suggestions'
+
+import AdditionalStats from './AdditionalStats'
+
+const DHIT_MOD = 1.25
+const TRAIT_STRENGTH = 0.20
+const APEX_POTENCY_THRESHOLDS = {
+	singleTarget: 255, // below this is worse than a BS/RA
+	multiTarget: 189, // below this is worse than a QN (+ BL reset)
+}
+
+export default class ApexArrow extends Module {
+	static handle = 'Apex'
+	static title = t('brd.apex.title')`Apex Arrow`
+	static debug = false
+
+	@dependency private stats!: AdditionalStats
+	@dependency private suggestions!: Suggestions
+
+	private ghostedApexCasts: number = 0
+	private badApexCasts: number = 0
+
+	protected init() {
+		this.addEventHook('normaliseddamage', {by: 'player', abilityId: ACTIONS.APEX_ARROW.id}, this.onApex)
```

Comment:
I recommend calling 
`	@dependency private data!: Data`
and then using
`this.data.actions.APEX_ARROW.id` for all uses of ACTIONS.xyz in order to start supporting layered data. Especially since this is a new module, better to keep up!

Reply:
Thanks, I didn't even know this was necessary in order to make use of the layers.